edx/edx-ora2
# [BB-2961] [BD-05] [TNL-7577] Added a templating mixin

This PR helps to interact with Studio API to give a list of presets

**JIRA**: [TNL-7577](https://openedx.atlassian.net/browse/TNL-7577)

**Discussions**: https://openedx.atlassian.net/browse/TNL-7577

**Dependencies**: https://github.com/edx/edx-platform/pull/25012

**Sandbox URL**: [Sandbox](https://studio.pr25012.sandbox.opencraft.hosting/) 

**Testing instructions**:

You should have a working devstack for this

1. Clone this repo in ~/dev_root/src
2.  Dev up lms_studio by `make dev.up.lms+studio`
3. Run:

```
docker exec -t edx.devstack.lms bash -c '. /edx/app/edxapp/venvs/edxapp/bin/activate && cd /edx/app/edxapp/edx-platform && pip uninstall -y ora2 && pip install -e /edx/src/edx-ora2 && pip freeze | grep ora2'


docker exec -t edx.devstack.studio bash -c '. /edx/app/edxapp/venvs/edxapp/bin/activate && cd /edx/app/edxapp/edx-platform && pip uninstall -y ora2 && pip install -e /edx/src/edx-ora2 && pip freeze | grep ora2'


```

This is for master branch but docker name might vary depending on which branch you are using.

4. Sign In to `Studio` edit a course and unit now when you click on `Open Response` button you will see a pop-up
5. The other pop-is are dummy but the first-option works.


**Author notes and concerns**:

~The presets are somewhat a dummy yet, but mostly the inspiration is taken from how `Problems` and `HTML` module~
~works, specially this https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/x_module.py#L1037~
~class and it used in HTML~

~https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/html_module.py#L49~

~I was not sure about the translation running using django and it's integration with Studio, couldn't get that
to work.~



**Screenshot**:
![image](https://user-images.githubusercontent.com/7670449/94591068-30695180-02a5-11eb-8193-d1f7d9530f89.png)


![image](https://user-images.githubusercontent.com/7670449/94591154-4b3bc600-02a5-11eb-8629-3e33490919f0.png)



**Reviewers**
